### PR TITLE
Create submodule files in LatLon

### PR DESCRIPTION
**Pull request by JulesKouatchou**
## Types of change(s)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Trivial change (affects only documentation or cleanup)

## Checklist
- [ ] Tested this change with a run of GEOSgcm
- [x] Ran the Unit Tests (`make tests`)

## Description
- Rename the folder `latlon` (in `geom_mgr) as `LatLon`.
- Create submodule files in `LatLon`.
- Change the version of `ESMA_cmake` from 3.46.0 to 3.37.0.

### CODE REVIEW

**Comment by JulesKouatchou** [comment]
Made a mistake in the ESMA_cmake version number. It should have been 3.47.0.

File: geom_mgr/LatLon/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  LatLonGeomSpec.F90
+  LatLonGeomFactory.F90
+
+)
+
+esma_add_fortran_submodules(
+  TARGET MAPL.geom_mgr
+  SUBDIRECTORY LatLonDecomposition
+  SOURCES new_LatLonDecomposition_basic.F90 new_LatLonDecomposition_petcount.F90
+          new_LatLonDecomposition_topo.F90 get_lon_distribution.F90
```
Comment (tclune) [inline geom_mgr/LatLon/CMakeLists.txt:15]: As I mentioned in Teams - the constructors should stay in the module and not be split as submodules.

File: geom_mgr/LatLon/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  SOURCES equal_to.F90 make_decomposition.F90 new_LatLonGeomSpec.F90
+          get_decomposition.F90 make_distribution.F90 supports_hconfig.F90
+          get_lat_axis.F90 make_LatLonGeomSpec_from_hconfig.F90
+          supports_metadata.F90 get_lon_axis.F90
+          make_LatLonGeomSpec_from_metadata.F90)
+
+esma_add_fortran_submodules(
+  TARGET MAPL.geom_mgr
+  SUBDIRECTORY LatAxis
+  SOURCES new_LatAxis.F90 equal_to.F90 not_equal_to.F90 supports_hconfig.F90
```
Comment (tclune) [inline geom_mgr/LatLon/CMakeLists.txt:41]: put constructors back in module

File: geom_mgr/LatLon/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+          supports_metadata.F90 make_LatAxis_from_hconfig.F90 
+          make_lataxis_from_metadata.F90 get_lat_range.F90 get_lat_corners.F90 
+          fix_bad_pole.F90)
+
+esma_add_fortran_submodules(
+  TARGET MAPL.geom_mgr
+  SUBDIRECTORY LonAxis
+  SOURCES equal_to.F90 get_lon_range.F90 make_LonAxis_from_metadata.F90
+          supports_hconfig.F90 get_lon_corners.F90 make_LonAxis_from_hconfig.F90
+          new_LonAxis.F90 supports_metadata.F90)
```
Comment (tclune) [inline geom_mgr/LatLon/CMakeLists.txt:51]: constructors go in module

**Review by tclune — CHANGES_REQUESTED** [review]
Constructors should stay in their original location.  Generally these are functions that start with the `new_` prefix.

Otherwise is fine.

**Comment by JulesKouatchou** [comment]
Ok. I will make the changes right now. I will run tests before I push the changes.

**Comment by JulesKouatchou** [comment]
Moved the constructors back to module files.

**Review by tclune — APPROVED** [review]

**Comment by mathomp4** [comment]
Hmm. Seems like a file is missing:

```
CMake Error at ESMA_cmake/esma_support/esma_add_*******_submodules.cmake:23 (target_sources):
  Cannot find source file:

    /root/project/workspace/build-MAPL/geom_mgr/LatLon/LatLonDecomposition_get_lon_distribution.F90
```

**Comment by tclune** [comment]
It is on the main branch of ESMA_cmake (just checked).  Maybe we need a CI update in this PR?

**Comment by mathomp4** [comment]
> It is on the main branch of ESMA_cmake (just checked). Maybe we need a CI update in this PR?

It should work. I pressed the update branch button just now to see.

**Comment by JulesKouatchou** [comment]
I will close this PR. I will make a few edits to the branch and submit a new PR.

**Pull request closed without merging**